build.gradle

```diff
@@ -224,6 +224,36 @@ subprojects {
             }
         }
     }
+
+
+    task copyDependencies(type: Copy) {
+        from configurations.runtime
+        into new File(buildDir, "dex")
+    }
+
+    task copyLib(type: Copy, dependsOn: copyDependencies) {
+        from new File(buildDir, "libs")
+        into new File(buildDir, "dex")
+    }
+
+    task dx(type: Exec, dependsOn: copyLib) {
+        workingDir new File(buildDir, "dex")
+        commandLine 'bash', '-e', '-c', """
+            ls -1 . > dexInputList.txt
+            dx.exec --dex --statistics --num-threads=4 --core-library --output=./file.dex --input-list=dexInputList.txt >dex.log 2>&1
+            grep 'method id:.*' dex.log
+        """.replace("dx.exec",(System.getProperty("dx.exec") == null) ? "dx" : System.getProperty("dx.exec"))
+    }
+
+    task deleteDex(type: Delete, dependsOn: dx) {
```

Comment:
I think you drop the `deleteDex` class and rely on the `clean` targets doing the right thing.